Samples/ios/SampleAppiOS-Swift/SampleAppiOS-Swift/SamplePhotoUtil.swift

```diff
@@ -0,0 +1,235 @@
+    }
+    
+    func cachedImagePath() -> String? {
+        if let currentUserIdentifier = SampleMSALUtil.shared.currentUserIdentifier {
+            return cachedImageDirectory() + "/" + currentUserIdentifier
+        }
+        return nil
+    }
+    
+    func setLastChecked() {
+        UserDefaults.standard.set(Data(), forKey: kLastPhotoCheckKey)
+    }
+    
+    func cache(photo data: Data) throws {
+        let imageDirectory = cachedImageDirectory()
+        
+        do {
+            if (FileManager.default.fileExists(atPath: imageDirectory) == false) {
+                try FileManager.default.createDirectory(atPath: imageDirectory, withIntermediateDirectories: true, attributes: [:])
+            }
+            
+            guard let imagePath = cachedImagePath() else {
+                throw SampleAppError.NoUserSignedIn
+            }
+            
+            try data.write(to: URL(fileURLWithPath: imagePath))
+        } catch let error {
+            throw SampleAppError.ImageCacheError(error)
+        }
+    }
+    
+    func checkTimestamp() -> Bool {
+        guard let cachedImagePath = cachedImagePath(),
+            let lastChecked = UserDefaults.standard.object(forKey: kLastPhotoCheckKey) as? Date else {
+            return true
+        }
+        
+        let cachedFileExists = FileManager.default.fileExists(atPath: cachedImagePath)
+        if (cachedFileExists) {
+            return (-lastChecked.timeIntervalSinceNow > kSecondsPerDay * 7)
+        }
+        else {
+            return (-lastChecked.timeIntervalSinceNow > kSecondsPerDay)
+        }
+    }
+    
+}
+
+// MARK: Request
+fileprivate extension SamplePhotoUtil {
+    
+    func getUserPhotoImpl(with completion: @escaping PhotoCompletion) {
+        // When acquiring a token for a specific purpose you should limit the scopes
+        // you ask for to just the ones needed for that operation. A user or admin might not
+        // consent to all of the scopes asked for, and core application functionality should
+        // not be blocked on not having consent for edge features.
+        let scopesRequired = [GraphScopes.UserRead.rawValue];
+        
+        SampleMSALUtil.shared.acquireTokenForCurrentUser(forScopes: scopesRequired) {
+            (token, error) in
+            
+            guard let accessToken = token else {
+                completion(nil, error)
+                return
+            }
+            
+            self.getPhoto(withToken: accessToken, completion: completion)
+        }
+    }
+    
+    
+    func getPhoto(withToken accessToken: String, completion: @escaping PhotoCompletion) {
+        let request = SampleGraphRequest(withToken: accessToken)
+        
+        getPhotoData(withRequest: request) {
+            (data, error) in
+            
+            if let error = error {
+                completion(nil, error)
+                return
+            }
+            
+            self.setLastChecked()
```

Comment:
Just a question: if a user doesn't have any photo (like me), we won't save this checking result and will keep rechecking for that user. Is it expected?